astropy/stats/sigma_clipping.py

```diff
@@ -455,7 +496,14 @@ def _sigmaclip_noaxis(self, data, masked=True, return_bounds=False, copy=True):
             return filtered_data
 
     def _sigmaclip_withaxis(
-        self, data, axis=None, masked=True, return_bounds=False, copy=True
+        self,
+        data: ArrayLike,
+        axis: int | tuple[int, ...] | None = None,
+        masked: bool | None = True,
+        return_bounds: bool | None = False,
+        copy: bool | None = True,
+    ) -> (
+        ArrayLike | tuple[ArrayLike, float, float] | tuple[ArrayLike, NDArray, NDArray]
```

Comment:
It's not actually ArrayLike, it's NDArray | MaskedArray.

Reply:
Fixed in [70b2007](https://github.com/astropy/astropy/pull/16562/commits/70b2007960c680ca2b474f8274f008b1f2030165)